### ISSUE TITLE
Port rotated_local() as a method to Godot 3.x along with addtional math helper functions present in Godot 4.x 

### ISSUE THREAD

**Issue by AlonJu** [godotengine/godot-proposals#10634]
### Describe the project you are working on

Working on a resident evil clone.

### Describe the problem or limitation you are having in your project

rotated() wouldn't allow me to transform the model on a local axis, but rotated_local worked for me

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

having more math helper functions would make it more usable

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

taking inspiration from the code in 4.x, I have made my own implementation as an autoload function 

```gdscript
func rotated_local(select:Spatial,axis:Vector3, angle:float)->Transform:
	var curTransform:Transform = select.transform
	return Transform(curTransform.basis * Basis(axis,angle), curTransform.origin)

func rotated_global(select:Spatial,axis:Vector3, angle:float)->Transform:
	var curTransform:Transform = select.transform
	return Transform(Basis(axis,angle)*curTransform.basis , Basis(axis,angle).xform(curTransform.origin))
```

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It can with the Autoloads script, but it isn't convenient, feels clunky and doesn't have documentation

### Is there a reason why this should be core and not an add-on in the asset library?

4.x has it already and it is already written there
https://docs.godotengine.org/en/stable/classes/class_transform3d.html#class-transform3d-method-rotated-local

**Comment by AThousandShips** [comment]
The workaround is trivial, by just multiplying by the other order, as per the documentation, you don't need a global method unless you want a minor optimization

**Comment by AlonJu** [comment]
@AThousandShips  I apologize, it was meant to be an example. I don't want a rotated_global() method, as rotated() already exists. I just desire a rotated_local() one as it is very useful for doing game dev in godot. It solved a problem that I wasn't able to do with the default helper function.

I also thank you for editing my post.

**Comment by AThousandShips** [comment]
No I mean that `rotated_local` is just `T * R`, the helper function is just marginally faster

**Comment by AlonJu** [comment]
That is true. While it is a minimal change to write, it is a very useful helper function. I embarrassingly got stuck on 3D projects for years until I looked into the source code for 4.x and added that into 3.x.

There are some beginners who want to use 4.x, but can't because their gpu is incompatible with Vulkan. So I think that it would be more accessible to them that way.
